frontend/components/App/App.tsx

```diff
@@ -73,6 +63,48 @@ const App = ({ children, location }: IAppProps): JSX.Element => {
 
   const [isLoading, setIsLoading] = useState(false);
 
+  // We will do a series of API calls to get the data that we need to display
+  // warnings to the user about various token expirations.
```

Comment:
This makes we wonder if there's a good way to expose a callback hook that any child component could use to refetch all the MDM credentials on demand that also handles updating the context. Just food for thought, not necessary to address in this PR.

Other ideas that come to mind would be to add some values from the location to the query keys, which we could use to trigger API calls to refetch based on the new location. It could be subject to a generous stale time to reduce the frequency of calls. Or it could be combined with a background refetch interval.

Reply:
> This makes we wonder if there's a good way to expose a callback hook that any child component could use to refetch all the MDM credentials on demand that also handles updating the context. Just food for thought, not necessary to address in this PR.

Yes we definitely could pull this out into a hook and reuse use this. that actually may be something we want to do when a user takes an action to resolve any of these warning.

Reply:
> Other ideas that come to mind would be to add some values from the location to the query keys, which we could use to trigger API calls to refetch based on the new location. It could be subject to a generous stale time to reduce the frequency of calls. Or it could be combined with a background refetch interval.

Yes I thought about this too. I think the ideal is recalling these explicitly when a user takes an action to resolve the issue (e.g. uploads a new vpp token --> refetch the data.) that way we are only calling when we know the data has been updated.